pkg/apis/unikorn/v1alpha1/types.go

```diff
@@ -680,3 +680,119 @@ type OpenstackSecurityGroupRuleSpec struct {
 
 type OpenstackSecurityGroupRuleStatus struct {
 }
+
+// ServerList is a typed list of servers.
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+type ServerList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []Server `json:"items"`
+}
+
+// Server defines a server beloning to an identity.
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:resource:scope=Namespaced,categories=unikorn
+// +kubebuilder:subresource:status
+// +kubebuilder:printcolumn:name="status",type="string",JSONPath=".status.conditions[?(@.type==\"Available\")].reason"
+// +kubebuilder:printcolumn:name="age",type="date",JSONPath=".metadata.creationTimestamp"
+// +kubebuilder:printcolumn:name="publicIP",type="string",JSONPath=".status.publicIP"
+type Server struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Spec              ServerSpec   `json:"spec"`
+	Status            ServerStatus `json:"status,omitempty"`
+}
+
+type ServerSpec struct {
```

Comment:
Good, no SSH config yet, as this is a task that involves Kubernetes clusters too we should do them together when actually needed!